Add documentation for derived source in source field metadata (#10674)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/index-settings.md

- `index.routing.allocation.total_primary_shards_per_node` (Integer): The maximum number of primary shards from a single index that can be allocated to a single node. This setting is applicable only for remote-backed clusters. Default is `-1` (unlimited). Helps control per-index primary shard distribution across nodes by limiting the number of primary shards per node. Use with caution because primary shards from this index may remain unallocated if nodes reach their configured limits.

> - `index.derived_source.translog.enabled` (Boolean): This setting controls, how to read documents from translog for derived source enabled index, defaults to `index.derived_source.enabled`. For more information, see [Derived source]({{site.url}}{{site.baseurl}}/field-types/metadata-fields/source/#derived-source).

Reviewer: ```suggestion
- `index.derived_source.translog.enabled` (Boolean): Controls how documents are read from the translog for an index with derived source enabled. Defaults to the `index.derived_source.enabled` value. For more information, see [Derived source]({{site.url}}{{site.baseurl}}/field-types/metadata-fields/source/#derived-source).
```
